code/modules/mechanics/MechanicMadness.dm

```diff
@@ -3291,7 +3293,120 @@
 		src.display_letter = new_letter
 		src.icon_state = new_icon_state
 
+/// allows cabinets to move around
+/obj/item/mechanics/movement
+	name = "Movement Component"
+	desc = "Allows a cabinet to move around."
+	icon_state = "comp_move"
+	cooldown_time = 1 SECOND
+	cabinet_only = TRUE
+	one_per_tile = TRUE
+	var/move_lag = BASE_SPEED
 
+	New()
+		..()
+		SEND_SIGNAL(src, COMSIG_MECHCOMP_ADD_INPUT, "walk", .proc/do_walk)
+		SEND_SIGNAL(src, COMSIG_MECHCOMP_ADD_INPUT, "step", .proc/do_step)
+		SEND_SIGNAL(src, COMSIG_MECHCOMP_ADD_CONFIG, "Set walk delay", .proc/set_speed)
+
+	secure()
+		if (istype(src.loc, /obj/item/storage/mechanics/housing_handheld))
+			src.loc.AddComponent(/datum/component/legs/four)
+		else
+			src.loc.AddComponent(/datum/component/legs/six)
+		RegisterSignal(src.loc, COMSIG_MOVABLE_SET_LOC, .proc/container_check)
+
+	loosen()
+		var/datum/component/C
+		if (istype(src.loc, /obj/item/storage/mechanics/housing_handheld))
+			C = src.loc.GetComponent(/datum/component/legs/four)
+		else
+			C = src.loc.GetComponent(/datum/component/legs/six)
+		if (C)
+			C.RemoveComponent()
+		src.stop_moving()
+		UnregisterSignal(src.loc, COMSIG_MOVABLE_SET_LOC)
+
+	/// checks if we are in a container so we can stop walking
+	proc/container_check()
+		var/obj/item/storage/S = src.loc
+		if (!istype(S))
+			return
+		if (!isturf(S.loc))
+			stop_moving()
```

Comment:
I'm not sure what is desired behaviour but should moving return when we are on a turf again?

If so you will likely need to store last walk arguments.

Reply:
No, I think it's fine if it stops!

Reply:
Intended behavior for me is to have it stop while in a container, so I think this is fine!